components/script/dom/globalscope.rs

```diff
@@ -3198,11 +3214,15 @@ impl GlobalScope {
                 task!(update_gamepad_state: move || {
                     let global = this.root();
                     if let Some(window) = global.downcast::<Window>() {
-                        let gamepad_list = window.Navigator().gamepads();
-                        if let Some(gamepad) = gamepad_list.Item(index as u32) {
+                        let navigator = window.Navigator();
+                        let gamepad_list = navigator.gamepads();
+                        if let Some(gamepad_or_none) = gamepad_list.get(index) {
```

Comment:
By the way are we sure the `GamepadIndex` will always match the index in the gamepad list? I guess because of the handling of a previous connected event? How would that play out if the focused webview changes?